Create Smart Meter

Smightym8 opened the pull request:
- **feat(auth): setup JWT authentication for API security**
  - Added middleware to handle token validation
  - Tested with a sample GET endpoint
  

- **feat(smartmeter): implement create smart meter**
  - Added CreateSmartMeter endpoint in SmartMeterController with openapi documentation
  - Added SmartMeterCreateDto
  - Added ISmartMeterCreateService and implemented creating smart meter in SmartMeterCreateService
  - Added ISmartMeterRepository with NextIdentity and AddAsync
  - Implemented NextIdentity and AddAsync in SmartMeterRepository
  - Removed many to many mapping between Policy and SmartMeter
  - Created UserNotFoundException and added it to GlobalExceptionHandler
  - Added integration tests to verify that a smart meter is created and that an access token is needed
  

- **feat(domain): add PolicySmartMeter entity to configure many-to-many relationship**
  - Added PolicySmartMeter entity to serve as a join table for the many-to-many relationship between Policy and SmartMeter entities.
  - Configured EF Core relationships.
  

- **refactor(smartmeter): change AddSmartMeterAsync to accept userId as string**
  

- **test(smartmeter): add unit tests for AddSmartMeterAsync method**
  

- **style: format code using dotnet format**

Smightym8 committed:
feat(auth): setup JWT authentication for API security

- Added middleware to handle token validation
- Tested with a sample GET endpoint

Smightym8 committed:
feat(smartmeter): implement create smart meter

- Added CreateSmartMeter endpoint in SmartMeterController with openapi documentation
- Added SmartMeterCreateDto
- Added ISmartMeterCreateService and implemented creating smart meter in SmartMeterCreateService
- Added ISmartMeterRepository with NextIdentity and AddAsync
- Implemented NextIdentity and AddAsync in SmartMeterRepository
- Removed many to many mapping between Policy and SmartMeter
- Created UserNotFoundException and added it to GlobalExceptionHandler
- Added integration tests to verify that a smart meter is created and that an access token is needed

Smightym8 committed:
feat(domain): add PolicySmartMeter entity to configure many-to-many relationship

- Added PolicySmartMeter entity to serve as a join table for the many-to-many relationship between Policy and SmartMeter entities.
- Configured EF Core relationships.

Smightym8 committed:
refactor(smartmeter): change AddSmartMeterAsync to accept userId as string

Smightym8 committed:
test(smartmeter): add unit tests for AddSmartMeterAsync method

Smightym8 committed:
style: format code using dotnet format

iiNomad23 commented on src/SMAIAXBackend.Domain/Repositories/IUserRepository.cs:
```diff
@@ -7,4 +7,5 @@ public interface IUserRepository
 {
     UserId NextIdentity();
     Task AddAsync(User user);
+    Task<User?> GetByIdAsync(UserId userId);
```
Did you mean 'GetUserByIdAsync'?

iiNomad23 reviewed (approved):
lgtm but check my comment

Smightym8 committed:
refactor(user): rename GetByIdAsync to GetUserByIdAsync in IUserRepository

- Renamed GetById to GetUserById in IUserRepository to improve clarity and specificity.
- Updated all references to GetById across the codebase to match the new method name.

Smightym8 committed:
Merge remote-tracking branch 'origin/main' into feature/create-smart-meter

# Conflicts:
#	src/SMAIAXBackend.API/Program.cs

Smightym8 committed:
style: format code using dotnet format
